src/main.js

```diff
@@ -11,18 +11,22 @@ Vue.use(VueAxios, axios);
 axios.defaults.baseURL = "https://developers.zomato.com/api/v2.1";
 axios.defaults.headers.common["user-key"] = process.env.VUE_APP_API_KEY;
 
-//interceptors
-
 axios.interceptors.response.use(
   response => {
-    if (response.status === 403) {
-      console.log(response);
+    if (process.env.VUE_APP_API_KEY) {
+      console.log(response.status);
     }
     return response;
   },
   error => {
-    if (error.response.status === 403) {
-      alert("please use your API key to run the app");
+    if (!process.env.VUE_APP_API_KEY) {
+      alert(
+        "Please make a file named '.env.local'. Create a global variable VUE_APP_API_KEY and initialize it with your own key in this file. " +
+          error
+      );
```

Comment:
Much better, the solution is almost what I'd expect, there are very few "buts":
* "create a global variable", "initialize" --- what if I'm not a technical person? I understand "create/make a file", but what is "create a global variable (...) and initialize it (...) in your file"?
* I followed the instructions, closed the window... and nothing happens. Reloaded. Nothing happens. If I wasn't a technical person, it's not a given I'd think of restarting the server just to update the key.
* you could provide a link to documentation, NEVER be afraid of it
* you don't SAY what the actual error was, just provide a solution and the "cause" error 403 --- I, as a user, don't really know why I should do that
* are you sure showing the "cause" error is needed at all here? The server did respond, so the network connection is fine and 403 (Forbidden) is what you get on authorization fails --- but you do know the cause of the authorization failure. The other alert (a few lines below) is much better, it uses the fact you already know the key exists, so if it still yielded 403 then the token may be invalid or expired, but certainly not absent.

All of that being said, `alert` isn't a really nice way to show errors like that, as it "breaks the wall" of your app. E.g. if your app is all bluish-greenish, and the alert window is black/dark-gray (in the browser I'm using to type this comment --- it is), then it feels off and as if it was the browser-related issue, not page reporting some casual page-related problem.